cpp/src/reshape/explode.cu

```diff
@@ -67,39 +68,41 @@ std::unique_ptr<table> explode_functor::operator()<list_view>(
    explode column. This unrolls the top level of lists. Then we need to insert the explode column
    back into the table and return it. */
   lists_column_view lc{input_table.column(explode_column_idx)};
-  rmm::device_uvector<size_type> gather_map_indices(lc.child().size(), stream, mr);
-  auto offsets = lc.offsets();
+  auto sliced_child = lc.get_sliced_child(stream);
+  rmm::device_uvector<size_type> gather_map_indices(sliced_child.size(), stream, mr);
 
-  auto offsets_minus_one = thrust::make_transform_iterator(offsets.begin<size_type>(),
-                                                           [] __device__(auto i) { return i - 1; });
-  auto counting_iter     = thrust::make_counting_iterator(0);
+  // sliced columns can make this a little tricky. We have to start iterating at the start of the
+  // offsets for this column, which could be > 0. Then we also have to handle rebasing the offsets
+  // as we go.
+  auto offsets           = lc.offsets().begin<size_type>() + lc.offset();
+  auto offsets_minus_one = thrust::make_transform_iterator(
+    offsets, [offsets] __device__(auto i) { return i - offsets[0] - 1; });
```

Comment:
```suggestion
    offsets, [offsets] __device__(auto i) { return (i - offsets[0]) - 1; });
```
I'm kind of a fan of being explicit with parentheses. Subtraction freaks me out.